src/util/gen-type-script.js

```diff
@@ -36,11 +38,13 @@ export function generateTypeScript () {
     for (let propName in builder._propertyDescriptors) {
       const propertyDescriptor = builder._propertyDescriptors[propName];
       if (propertyDescriptor.Name !== propName) {
-        throw `Mismatched prop name, ${elementName}.${propName} !== ${propertyDescriptor.Name}`;
```

Comment:
@grozdanov, can you explain the difference between the key into _propertyDescriptors and the Name property on the property descriptor object itself? Previously I was throwing an exception if they didn't match while generating TypeScript (since I wasn't sure which one I should be using otherwise -- and in practice they all _did_ match). This changed with Prism where there are some differences (e.g. key = 'setHandGestureHoverDistance', Name = 'handGestureHoverDistance', and several others with same pattern).

Is this intentional and, if so, which name corresponds to the actual property name I should use for TypeScript?

Reply:
It is not intentional 😞. It is a bug 🐞 . Thank you @tonycuadra !

Reply:
Back to throwing an exception then (since it will help catch anything like this in the future). Too easy to miss a console log.